src/compiler/checker.ts

```diff
@@ -16230,6 +16230,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return type.elementFlags.length - findLastIndex(type.elementFlags, f => !(f & flags)) - 1;
     }
 
+    function getTotalFixedElementCount(type: TupleType) {
+        return type.fixedLength + getEndElementCount(type, ElementFlags.Fixed);

```

Comment:
Does every use of `fixedLength` go through this function now? If so, does that imply that this property should be set to this value instead? Or that other uses are also subtly wrong?

Reply:
> Does every use of fixedLength go through this function now? If so, does that imply that this property should be set to this value instead?

No, many call sites are interested in the **leading** fixed length so they are not wrong right now and they need the value that is currently held in this property.

> Or that other uses are also subtly wrong?

Some places might still be subtly wrong. I'll likely investigate this if I don't forget but that likely can be just done in followup PRs.